File: fix_encoding.js
const fs = require('fs');
const path = require('path');

const files = fs.readdirSync(__dirname).filter(f => f.endsWith('.html') && f !== 'index.html');

for (const file of files) {
    let content = fs.readFileSync(file, 'utf8');

    // Fix the variables that got messed up
    content = content.replace(/C[?\uFFFD]digos/gi, 'codigos');
    content = content.replace(/c[?\uFFFD]digo/gi, 'codigo');
    content = content.replace(/cdigo/gi, 'codigo');
    content = content.replace(/Cdigo/gi, 'codigo');

    // Restore text content correctly
    content = content.replace(/placeholder=".*? aqui"/g, 'placeholder="Código aqui"');
    content = content.replace(/<th>C.*?digo<\/th>/gi, '<th>Código</th>');
    content = content.replace(/<h4 class="mb-3">.*? Total de/g, '<h4 class="mb-3">📦 Total de');
    content = content.replace(/">.*? Exportar Excel/g, '">📊 Exportar Excel');
    content = content.replace(/">.*? Limpar Tudo/g, '">🗑️ Limpar Tudo');
    content = content.replace(/<th>.*O<\/th>/g, '<th>❌</th>');
    content = content.replace(/\/\/ .*? pegar hora atual/g, '// 🕒 pegar hora atual');
    content = content.replace(/salva c.*?digo \+ hora/g, 'salva código + hora');
    content = content.replace(/N.*?o c/g, 'Não c');
    content = content.replace(/n.*?o apenas/g, 'não apenas');
    content = content.replace(/<a href="index.html" class="botao-voltar">.*?<\/a>/g, '<a href="index.html" class="botao-voltar">⬅️</a>');

    content = content.replace(/Total de C.*?digos:/g, 'Total de Códigos:');
    content = content.replace(/'C.*?digo de Barras'/g, "'Código de Barras'");
    content = content.replace(/'C.*?digo'/g, "'Código'");
    content = content.replace(/Nenhum c.*?digo para/g, "Nenhum código para");
    content = content.replace(/remover este c.*?digo/g, "remover este código");
    content = content.replace(/todos os c.*?digos/g, "todos os códigos");
    
    fs.writeFileSync(file, content, 'utf8');
}
console.log('Fixed all HTML files.');

File: fix_encoding.py
import os
import glob
import re

for file in glob.glob("*.html"):
    if file == "index.html":
        continue
    with open(file, "r", encoding="utf-8", errors="ignore") as f:
        content = f.read()
    
    # Restoring the text
    content = re.sub(r'placeholder=".*? aqui"', 'placeholder="Código aqui"', content)
    content = re.sub(r'<th>C.*?digo</th>', '<th>Código</th>', content)
    content = re.sub(r'<h4 class="mb-3">.*? Total de', '<h4 class="mb-3">📦 Total de', content)
    content = re.sub(r'">.*? Exportar Excel', '">📊 Exportar Excel', content)
    content = re.sub(r'">.*? Limpar Tudo', '">🗑️ Limpar Tudo', content)
    content = re.sub(r'<th>.*O</th>', '<th>❌</th>', content)
    content = re.sub(r'// .*? pegar hora atual', '// 🕒 pegar hora atual', content)
    content = re.sub(r'salva c.*?digo \+ hora', 'salva código + hora', content)
    content = re.sub(r'N.*?o c', 'Não c', content)
    content = re.sub(r'n.*?o apenas', 'não apenas', content)
    content = re.sub(r'<a href="index\.html" class="botao-voltar">.*?</a>', '<a href="index.html" class="botao-voltar">⬅️</a>', content)
    content = re.sub(r'Total de C.*?digos:', 'Total de Códigos:', content)
    content = re.sub(r"'C.*?digo de Barras'", "'Código de Barras'", content)
    content = re.sub(r"'C.*?digo'", "'Código'", content)
    content = re.sub(r'Nenhum c.*?digo para', "Nenhum código para", content)
    content = re.sub(r'remover este c.*?digo', "remover este código", content)
    content = re.sub(r'todos os c.*?digos', "todos os códigos", content)
    content = re.sub(r'let c.*?digos', "let codigos", content)
    content = re.sub(r'let C.*?digos', "let codigos", content)
    
    # Fix variables like codigosA, codigosB
    content = re.sub(r'c.*?digos([a-zA-Z])', r'codigos\1', content)
    content = re.sub(r'C.*?digos([a-zA-Z])', r'codigos\1', content)
    content = re.sub(r'c.*?digo\b', r'codigo', content)
    content = re.sub(r'C.*?digo\b', r'codigo', content)

    # Some remaining variables might be exactly 'codigos' (no letter suffix)
    # The previous regex catches 'c.*?digos([a-zA-Z])' meaning it misses 'codigos'
    # Actually 'let c.*?digos' handles some of it.
    
    # Just generic variable name fix for 'codigos' when used as an array name
    content = content.replace("códigoA", "codigosA")
    content = content.replace("CódigoA", "codigosA")

    with open(file, "w", encoding="utf-8") as f:
        f.write(content)
